src/utilities.jl

```diff
@@ -0,0 +1,42 @@
+# high-level utilities
+
+macro collect(ex)
+    quote
+        isactive() || @warn("No ITT collector present. Make sure you have a collector attached to your process, e.g., by running under VTune.")
+        resume()
+        # Use Expr(:tryfinally, ...) so we don't introduce a new soft scope.
+        # TODO: switch to solution once https://github.com/JuliaLang/julia/pull/39217 is resolved
+        $(Expr(:tryfinally, esc(ex), :(pause())))
+    end
+end
+
+macro task(ex...)
+    length(ex) >= 2 || error("Usage: IntelITT.@task [domain=...] name::String code")
+    name = ex[end-1]
+    code = ex[end]
+    kwargs = ex[1:end-2]
+
+    # handle kwargs
+    domain = nothing
+    for kwarg in kwargs
+        key::Symbol, val = kwarg.args
+        if key === :domain
+            domain = val
+        else
+            error("unknown keyword argument: $key")
+        end
+    end
+    if domain === nothing
+        domain = :(Domain(string($__module__)))
+    end
+
+    domain
+
+    quote
+        _domain = $(esc(domain))
+        task_begin(_domain, $name)
```

Comment:
It might be better to use `__itt_task_begin_overlapped` / `__itt_task_end_overlapped` in case Julia switches tasks, e.g. if you have two Julia tasks with `@task` annotations then you may end up with
```
# Start Julia task 1
task_begin(domain, "task 1")
# switch to Julia task 2
task_begin(domain, "task 2")
# switch back to Julia task 1
task_end(domain) # task 1
# switch back to Julia task 2
task_end(domain) # task 2
```

NVTX also supports the case where the tasks thread-switch: it's not clear from the docs if ITT supports this?

Reply:
Good idea, I switched to the overlapped API.

> it's not clear from the docs if ITT supports this?

I'm not sure. In any case, not much we can do about this restriction from the IntelITT.jl side?

Reply:
> A task is a logical unit of work performed by a particular thread. Tasks can nest; thus, tasks typically correspond to functions, scopes, or a case block in a switch statement. You can use the Task API to assign tasks to threads.
>
> Task API is a per-thread function that works in resumed state. This function does not work in paused state. 

Looks pretty thread-specific.